developer/developer-guide.md

```diff
@@ -1,3 +1,872 @@
+* *apix-integrationtests* contains the integration tests for each Alfresco version.
+
+## How to
+
+### Run
+If it is the first time you build Alfred API on your machine:
+```bash
+./setup.sh  # or ./setup.bat on Windows
+```
+Then:
+```bash
+./gradlew :apix-docker:docker-${VERSION}:composeUp --info
+```
+Where `VERSION` is e.g. `51`.
+This starts up all docker containers required for an Alfresco running Alfred API.
+
+
+### Run integration tests
+```bash
+./gradlew :apix-integrationtests:test-${VERSION}:integrationTest
+```  
+Again, where `VERSION` is e.g. `51`.
+
+However, this starts (and afterwards stops) docker containers. This includes starting an Alfresco container,
+ adding a startup time of several minutes. To circumvent this you also run the test on already running containers with
+ for example:
+ ```bash
+./gradlew -x composeUp -x composeDown :apix-integrationtests:test-61:integrationTest -Pprotocol=http -Phost=localhost -Pport=8061
+```
+
+
+### Run integration tests under debugger
+1. Debugging settings are already added by `apix-docker/${VERSION}/debug-extension.docker-compose.yml`, including a 
+portmapping `8000:8000`. This file does not get loaded when running in Jenkins.
+2. Prepare your remote debugger in IntelliJ and set breakpoints where you want in your tests
+ (or Alfred API code).
+3. Run the integration tests (see section above).
+4. Wait until the container is started and healthy, then attach the debugger.
+
+Again, where `VERSION` is e.g. `51`.
+
+#### Deploy code changes for development
+In a development scenario, it is possible to upload code changes to a running alfresco through dynamic extensions.
+This requires the running alfresco to already have an older or equal version of alfred-api installed, and
+the use of the jar artifact instead of the amp to do the new install. 
+The JAR has the format `apix-impl-{ALFRESCO-VERSION}-{APIX-VERSION}.jar` and can be found under 
+`apix-impl/{ALFRESCO-VERSION}/build/libs/`, where `ALFRESCO-VERSION` is one of *(50|51|52|60|61|62)*.
+The new installation can be done either through the DE web interface, or with the following gradle task.
+```bash
+./gradlew :apix-impl:apix-impl-{ALFRESCO-VERSION}:installBundle -Phost={ALFRESCO-HOST} -Pport={ALFRESCO-PORT}
+```
+
+*Protip:* If you get tired of changing the port after every `docker-compose up`, you can temporarily put a
+fixed port in the *docker-compose.yml* of the version you are working with. (The rationale behind using 
+variable ephemeral ports is that during parallel builds on Jenkins port clashes must be avoided.)
+
+For example for version 5.1, change in *apix-docker/51/docker-compose.yml* 
+the ports line from:
+```yaml
+services:
+  alfresco-core:
+    ports:
+      - ${DOCKER_IP}:8080
+``` 
+to: 
+```yaml
+services:
+  alfresco-core:
+    ports:
+      - ${DOCKER_IP}:9051:8080
+```
+and then restart the containers with:
+
+```bash
+./gradlew :apix-docker:docker-51:composeUp --info
+```
+
+# Alfred api Concepts
```

Comment:
Since this is the 1st version of the official docs, bear with me since I want to make sure details are correct too :smiley: 
Let's ensure "API" (and other often used acronyms) is always consistently spelled upper-case `API`. So:

- `API`
- `Alfred API`
- `REST`
- `JSON`
- ... you get the idea

I suggest a search-and-replace.